kbfsgit/storer.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package kbfsgit
+
+import (
+	"github.com/keybase/kbfs/libfs"
+	gogitcfg "gopkg.in/src-d/go-git.v4/config"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+	"gopkg.in/src-d/go-git.v4/storage/filesystem"
+)
+
+// configInMemoryStorer keeps the git config in memory, to work around
+// a gcfg bug (used by go-git when reading configs from disk) that
+// causes a freakout when it sees backslashes in git file URLs.
+type configInMemoryStorer struct {
+	*filesystem.Storage
+	cfg *gogitcfg.Config
+}
+
+func newConfigInMemoryStorer(fs *libfs.FS) (*configInMemoryStorer, error) {
+	fsStorer, err := filesystem.NewStorage(fs)
```

Comment:
I don't think so -- whatever we make needs to implement the full `Storer` interface, because that's what gogit expects in the `Init` call.  Unless you see a way around that?